.github/workflows/pull-request.yml

```
@@ -0,0 +1,265 @@
+        if: failure()
+        run: npm run format
+
+      - name: Check for type errors
+        if: steps.changed-files.outputs.only_changed != 'true'
+        run: npm run typecheck
+
+      - name: Check for linting errors in modified files
+        if: steps.changed-files.outputs.only_changed != 'true'
+        env: 
+          CHANGED_FILES: ${{ steps.changed_files.outputs.all_changed_files }}
+        run: npx eslint ${CHANGED_FILES} && python .github/workflows/eslint_disable_check.py
```

actionlint (1.7.3)

shellcheck reported issue in this script: SC2086:info:1:12: Double quote to prevent globbing and word splitting

(shellcheck)